manual_code/lab2/TM/src/kernel/k_task.c

```diff
@@ -389,9 +389,14 @@ int k_tsk_create(task_t *task, void (*task_entry)(void), U8 prio, U16 stack_size
 
 void k_tsk_exit(void) 
 {
-#ifdef DEBUG_0
-    printf("k_tsk_exit: entering...\n\r");
-#endif /* DEBUG_0 */
+// #ifdef DEBUG_0
+//     printf("k_tsk_exit: entering...\n\r");
+// #endif /* DEBUG_0 */
+    gp_current_task->state = DORMANT;
+    
+    RTX_TASK_INFO *b;
+    k_tsk_get(gp_current_task->tid, b);
+    k_mem_dealloc(b->u_sp);
```

Comment:
The TCB struct is being modified to include the field storing the base of the user stack pointer, so we can use that to avoid calling k_tsk_get().

`k_mem_dealloc(gp_current_task->u_stack_hi);`

Reply:
updated. The TCB changes arent reflected in this branch or master yet so I'll just assume it works